src/L2VoteAggregator.sol

```diff
@@ -90,29 +111,103 @@ abstract contract L2VoteAggregator {
+
+  /// @notice This function does not make sense in the L2 context, but we have added it to have
+  /// compatibility with existing Governor tooling.
+  function getVotes(address, uint256) public view virtual returns (uint256) {
+    return 0;
+  }
+
+  /// @notice This function does not make sense in the L2 context, but we have added it to have
+  /// compatibility with existing Governor tooling.
+  function propose(address[] memory, uint256[] memory, bytes[] memory, string memory)
+    public
+    virtual
+    returns (uint256)
+  {
+    revert UnsupportedMethod();
+  }
+
+  /// @notice This function does not make sense in the L2 context, but we have added it to have
+  /// compatibility with existing Governor tooling.
+  function execute(address[] memory, uint256[] memory, bytes[] memory, bytes32)
+    public
+    payable
+    virtual
+    returns (uint256)
+  {
+    revert UnsupportedMethod();
+  }
+
   /// @notice Where a user can express their vote based on their L2 token voting power.
   /// @param proposalId The id of the proposal to vote on.
   /// @param support The type of vote to cast.
   function castVote(uint256 proposalId, VoteType support) public returns (uint256) {
-    if (!proposalVoteActive(proposalId)) revert ProposalInactive();
-    if (_proposalVotersHasVoted[proposalId][msg.sender]) revert AlreadyVoted();
-    _proposalVotersHasVoted[proposalId][msg.sender] = true;
+    return _castVote(proposalId, msg.sender, support, "");
+  }
 
-    L2GovernorMetadata.Proposal memory proposal = GOVERNOR_METADATA.getProposal(proposalId);
-    uint256 weight = VOTING_TOKEN.getPastVotes(msg.sender, proposal.voteStart);
-    if (weight == 0) revert NoWeight();
+  /// @notice Where a user can express their vote based on their L2 token voting power, and provide
+  /// a reason.
+  /// @param proposalId The id of the proposal to vote on.
+  /// @param support The type of vote to cast.
+  /// @param reason The reason the vote was cast.
+  function castVoteWithReason(uint256 proposalId, VoteType support, string calldata reason)
+    public
+    virtual
+    returns (uint256)
+  {
+    return _castVote(proposalId, msg.sender, support, reason);
+  }
 
-    if (support == VoteType.Against) {
-      proposalVotes[proposalId].againstVotes += SafeCast.toUint128(weight);
-    } else if (support == VoteType.For) {
-      proposalVotes[proposalId].forVotes += SafeCast.toUint128(weight);
-    } else if (support == VoteType.Abstain) {
-      proposalVotes[proposalId].abstainVotes += SafeCast.toUint128(weight);
-    } else {
-      revert InvalidVoteType();
-    }
-    emit VoteCast(msg.sender, proposalId, support, weight);
-    return weight;
+  /// @notice Where a user can express their vote based on their L2 token voting power using  a
+  /// signature.
+  /// @param proposalId The id of the proposal to vote on.
+  /// @param support The type of vote to cast.
+  function castVoteBySig(uint256 proposalId, VoteType support, uint8 v, bytes32 r, bytes32 s)
```

Comment:
I just noticed we don't have the `castVoteWithReasonAndParamsBySig` method. Tally does not list that on their docs—I wonder why not? Would it be hard to include now that we've got the rest of them in place, just for consistency with the standard Governor?

Reply:
I think it should be pretty straightforward to add. I will give it a shot.

Reply:
Actually, I think this is best to split into a separate issue. My understanding is it should be pretty much the same as what we have in Flexible Voting [here](https://github.com/ScopeLift/flexible-voting/blame/master/src/GovernorCountingFractional.sol#L269). But, I don't want to hold this pr up while trying to implement.

Reply:
Yeah, let's discuss further. Thanks for pulling out the issue though 👍